beacon_node/beacon_chain/src/blob_verification.rs

```diff
@@ -201,25 +219,54 @@ pub fn validate_blob_sidecar_for_gossip<T: BeaconChainTypes>(
     //
     // However, we check that the proposer_index matches against the shuffling first to avoid
     // signature verification against an invalid proposer_index.
-    // TODO: check if getting the shuffling more expensive than signature verification in any scenario?
     let proposer_shuffling_root = signed_blob_sidecar.message.block_parent_root;
 
-    let (proposer_index, fork) = match chain
+    let proposer_opt = chain
         .beacon_proposer_cache
         .lock()
-        .get_slot::<T::EthSpec>(proposer_shuffling_root, blob_slot)
-    {
-        Some(proposer) => (proposer.index, proposer.fork),
-        None => {
-            let state = &chain.canonical_head.cached_head().snapshot.beacon_state;
+        .get_slot::<T::EthSpec>(proposer_shuffling_root, blob_slot);
+
+    let (proposer_index, fork) = if let Some(proposer) = proposer_opt {
+        (proposer.index, proposer.fork)
+    } else {
+        // The cached head state is in the same epoch as the blob or the state has already been
+        // advanced to the blob's epoch
+        let snapshot = &chain.canonical_head.cached_head().snapshot;
+        if snapshot.beacon_state.current_epoch() == blob_slot.epoch(T::EthSpec::slots_per_epoch()) {
             (
-                state.get_beacon_proposer_index(blob_slot, &chain.spec)?,
-                state.fork(),
+                snapshot
+                    .beacon_state
+                    .get_beacon_proposer_index(blob_slot, &chain.spec)?,
+                snapshot.beacon_state.fork(),
             )
         }
+        // Need to advance the state to get the proposer index
```

Comment:
At this point maybe it's just better to `IGNORE` for now and eventually add a smarter reprocessing logic? Adding a `QueuedGossipBlob` to the reprocessing queue?

It's possible we end up doing this in parallel for all blobs for a slot as we recieve them right?

Reply:
Talked offline, pawan's gonna leave as is for now because this should be rarely hit